.github/workflows/build-push-multi-arch.yml

```diff
@@ -0,0 +1,107 @@
+name: Build and Push (multi-arch)
+
+on:
+  workflow_call:
+    inputs:
+      image_name:
+        description: 'The name of the image (for example "aica-technology/image-name")'
+        required: true
+        type: string
+      image_tag:
+        description: 'The tag of the image (for example "latest")'
+        required: true
+        type: string
+      build_context_path:
+        description: 'Path to the directory that specifies the Docker build context'
+        required: true
+        type: string
+        default: '.'
+      dockerfile_path:
+        description: 'Path of the Dockerfile'
+        required: true
+        type: string
+        default: 'Dockerfile'
+      architecture:
+        description: 'The desired architecture [`amd64` or `amd64 & arm64`]'
+        required: true
+        type: string
+        default: 'amd64 & arm64'
+      build_flags:
+        description: 'Any build flags that should be forwarded to the build process as an array'
+        required: false
+        type: string
+      artifact_name:
+        description: 'Name of the artifact to be downloaded'
+        required: false
+        type: string
+      artifact_path:
+        description: 'Path where the artifact should be downloaded to'
+        required: false
+        type: string
+    secrets:
+      token:
+        description: 'The GitHub token passed from the caller workflow'
+        required: true
+    outputs:
+      image_hash:
+        description: 'The SHA of the pushed image'
+        value: ${{ jobs.build-publish.outputs.image_hash }}
+
+jobs:
+  build-publish:
+    runs-on: ubuntu-latest
+    name: Build and push image
+    outputs:
+      image_hash: ${{ steps.build-push.outputs.sha }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Download artifact
+        if: inputs.artifact_name
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.artifact_name }}
+          path: ${{ inputs.artifact_path }}
+
```

Comment:
Just to check: this is not currently used, but you placed it here to allow other jobs to share data if they have an `upload-artifacts` step?